vignettes/howtogenefilter.Rmd

```diff
@@ -0,0 +1,190 @@
+
+There are three steps that must be performed.
+
+1.  Create function(s) implementing the filtering criteria.
+
+2.  Assemble it (them) into a (combined) filtering function.
+
+3.  Apply the filtering function to the expression matrix.
+
+```{r}
+f1 <- kOverA(5, 200)
+ffun <- filterfun(f1)
+wh1 <- genefilter(exprs(sample.ExpressionSet), ffun)
+sum(wh1)
+```
+
+Here `f1` is a function that implies our "expression measure above 200
+in at least 5 samples" criterion, the function `ffun` is the filtering
+function (which in this case consists of only one criterion), and we
+apply it using `genefilter`. There were genes that satisfied the
+criterion and passed the filter.
+As an example for a specific filter, let us select genes that are
+differentially expressed in the groups defined by `type`.
+
+```{r}
+f2 <- ttest(sample.ExpressionSet$type, p=0.1)
+ffun <- filterfun(f1)
+wh2 <- genefilter(exprs(sample.ExpressionSet), 
+filterfun(f2))
+sum(wh2)
+```
+Here, `ttest` is a function from the `genefilter`
+package which provides a suitable wrapper around \texttt{t.test} from
+package \textit{stats}. Now we see that there are  `r \Sexpr{sum(wh2)}`
+genes that satisfy the selection criterion. 
+
+Suppose that we want to combine the two filters. We want those genes
+for which at least 5 have an expression measure over 200 *and* which also are differentially
+expressed between the groups defined by `type+`.
+
+```{r}
+ffun_combined <- filterfun(f1, f2)
+wh3 <- genefilter(exprs(sample.ExpressionSet), ffun_combined)
+sum(wh3)
+```
+
+Now we see that there are only \Sexpr{sum(wh3)} genes  that satisfy both conditions.
+
+FIXME: need to replace this with something else
+Our last example is to select genes that are
+differentially expressed in at least one of the three groups defined
+by \verb+cov3+.
+To do that we use an Anova filter. This filter uses an analysis of
+variance appraoch (via the \verb+lm+) function to test the hypothesis
+that at least one of the three group means is different from the other
+two. The test is applied, then the $p$--value computed. We select
+%those genes that have a low $p$--value.
+
+
+Afilter <- Anova(eset$cov3)
+aff <- filterfun(Afilter)
+wh4 <- genefilter(exprs(eset), aff)
+sum(wh4)
+
+We see that there are 14 genes that pass this filter and that are
+candidates for further exploration.
+
+\section*{Selecting genes that appear useful for prediction}
+
+The function \texttt{knnCV} defined below performs$k$--nearest neighbour classification
+using leave--one--out cross--validation.
+At the same time it aggregates the genes that were selected. The
+function returns the predicted classifications as its returned
+value. However, there is an additional side effect. The number of
+times that each gene was used (provided it was at least one) are
+recorded and stored in the environment of the aggregator \verb+Agg+.
+These can subsequently be retrieved and used for other purposes.
+
+```{r}
```

Comment:
Please add back in the names of code blocks. In this vignette, not all of the code blocks have name. It's good practice to name code blocks so let's keep the ones it has. You can the corresponding code block in the .Rnw at https://github.com/Bioconductor/genefilter/pull/11/files#diff-5f517776a2ea853f30900826151c07b985a403053eb75a6f6f417be9595e8c9cL138 to find its name. Please do this for other code blocks that have a name.

Reply:
This is still unresolved.